include/deal.II/base/utilities.h

```diff
@@ -1442,6 +1428,45 @@ namespace Utilities
                  allow_compression);
   }
 
+
+
+  template <typename Integer>
+  std::vector<Integer>
+  reverse_permutation(const std::vector<Integer> &permutation)
+  {
+    const std::size_t n = permutation.size();
+
+    std::vector<Integer> out(n);
+    for (std::size_t i = 0; i < n; ++i)
+      out[i] = n - 1 - permutation[i];
+
+    return out;
+  }
+
+
+
+  template <typename Integer>
+  std::vector<Integer>
+  invert_permutation(const std::vector<Integer> &permutation)
+  {
+    const std::size_t n = permutation.size();
+
+    std::vector<Integer> out(n, numbers::invalid_unsigned_int);
+
+    for (std::size_t i = 0; i < n; ++i)
+      {
+        AssertIndexRange(permutation[i], n);
+        out[permutation[i]] = i;
+      }
+
+    // check that we have actually reached
+    // all indices
+    for (unsigned int i = 0; i < n; ++i)
```

Comment:
Here is something I forgot:
```suggestion
    for (std::size_t i = 0; i < n; ++i)
```